clients/venice-client/src/main/java/com/linkedin/venice/fastclient/DispatchingAvroGenericStoreClient.java

```diff
@@ -176,6 +200,7 @@ protected CompletableFuture<V> get(GetRequestContext requestContext, K key) thro
                 ? HttpStatus.fromCode(((VeniceClientHttpException) throwable).getHttpStatus())
                 : HttpStatus.S_503_SERVICE_UNAVAILABLE;
             routeRequestFuture.complete(statusCode);
+            // TODO QQ: not completing valueFuture here? How this case will be handled?
```

Comment:
This is handled by the following `allOf` to wait for all the futures.

Reply:
Makes sense. Thanks.